### PR TITLE
Reapply "network: add "mac" to alternatives name policy by default"

### PR DESCRIPTION
**Pull request by yuwata**
This reverts commit 152c8946b3a93bb2c086568832d614cb54f11e13.

The kernel patch
https://lore.kernel.org/linux-usb/20240605153340.25694-1-gmazyland@gmail.com/ is now in net-next branch, and will be hopefully merged soon.

Note that Debian's 73-usb-net-by-mac.link now also supports the kernel patch: https://salsa.debian.org/systemd-team/systemd/-/commit/c1afbb2dc295929085be86072c7942c8517ec598

So, hopefully, the change is ready. Let's reapply it.

### CODE REVIEW

**Comment by bluca** [comment]
> Note that Debian's 73-usb-net-by-mac.link now also supports the kernel patch:

Should this be shipped upstream? There's no downstream patch at all, and it seems behaviour one generally wants for usb nics?

**Comment by yuwata** [comment]
> > Note that Debian's 73-usb-net-by-mac.link now also supports the kernel patch:
> 
> Should this be shipped upstream? There's no downstream patch at all, and it seems behaviour one generally wants for usb nics?

At least, the current form of 73-usb-net-by-mac.link should not be in upstream:
- For an external ethernet dongle, using `mac` with higher priority may not be bad, but an internal interface may be also connected to USB bus, like old Raspberry pi models, and in such case I cannot say `mac` based naming is better. Currently we have no way to check the interface is internal or external...
- Introducing another .link file like 73-usb-net-by-mac.link may change interface name. That's unacceptable for keeping backward compatibility unless the current 99-default.link really causes significant issues. Of course, we have versioning of naming scheme... But, I do not think there is enough reasons to introduce new flag about that.
- Nowadays, the kernel supports alternative names, so people can use `mac` based name with this patch (and without 73-usb-net-by-mac.link) without keeping the main name. At least, both `ip` command and `networkd` handle alternative names as same as the main name. So, I think there is no strong reason to swap the main name with `mac` based alternative name assigned by this patch.

In short, I cannot find any strong reason to use the `mac` based name as the main name.
